frame-metadata/src/v15.rs

```diff
@@ -249,11 +262,13 @@ impl IntoPortable for SignedExtensionMetadata {
 /// All metadata about an runtime pallet.
 #[derive(Clone, PartialEq, Eq, Encode, Debug)]
 #[cfg_attr(feature = "decode", derive(Decode))]
-#[cfg_attr(feature = "serde_full", derive(Serialize))]
 #[cfg_attr(
 	feature = "serde_full",
-	serde(bound(serialize = "T::Type: Serialize, T::String: Serialize"))
+	derive(Serialize),
+	serde(bound(serialize = "T::Type: Serialize, T::String: Serialize")),
+	serde(bound(deserialize = "T::Type: Deserialize<'de>, T::String: Deserialize<'de>"))
 )]
+#[cfg_attr(all(feature = "serde_full", feature = "decode"), derive(Deserialize))]
```

Comment:
Just let's only use `serde_full` everywhere. No need to also put this behind the `decode` feature.